train.py

```diff
@@ -101,11 +97,79 @@ def printTree(root):#this is for testing purposes
             print("Internal"+str(current.splitAttribute)+" "+str(current.threshold))
             open.append(current.leftChild)
             open.append(current.rightChild)
-        
-    
+            
+            
+def getBestInfoGain(data_cases,attrib):
```

Comment:
This function is called `getBestInfoGain()`, but the other is just `entropy()` and `infoGain()`. I might suggest getBestInfoGain, getEntropy, getInfoGain

If you want to be overly descriptive and make long function names, you might do getBestInfoGainForAttribute, getEntropyForSubset, and getInfoGainForThreshold. But that's just my overly-awkward style